File: lang/go/src/fun/vac/util/RandomFactory/RandomFactory.go
package RandomFactory

import (
    "math/rand"
    "time"
)

func Seed() {
    rand.Seed(time.Now().UnixNano())
}

func IntegerN(min int, max int) int {
    return min + rand.Intn(max - min + 1)
}

func GenerateInteger() int {
    return IntegerN(0, 2147483647)
}

func GenerateEven() int {
    return GenerateInteger() >> 1 << 1
}

func GenerateOdd() int {
    return GenerateInteger() | 1
}

File: lang/php/src/vac.fun/util/RandomFactory.php
<?php
    namespace vac\fun\util\RandomFactory;

    function seed(): void {
        // preserve for consistent interface
    }

    function integerN(int $min, int $max): int {
        return random_int($min, $max);
    }

    function generateInteger(): int {
        return integerN(0, 2147483647);
    }

    function generateEven(): int {
        return generateInteger() >> 1 << 1;
    }

    function generateOdd(): int {
        return generateInteger() | 1;
    }
?>
